File: Faza6/OnlineGamesShop/src/main/java/com/rich/onlinegamesshop/model/views/GameReviewsView.java
package com.rich.onlinegamesshop.model.views;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Immutable;

@Data
@Entity
@Immutable
@Table(name = "gamereviewsview")
@NoArgsConstructor
@AllArgsConstructor
public class GameReviewsView {
    @Id
    private Integer id_review;
    private Integer rating;
    private String comment;
    private String username;
    private Integer id_game;
    private String game_title;
    private String game_genre;
}

File: Faza6/FrontEnd/Angular/src/app/model/Views/GameReviewsView.ts
//TODO: make this a class model with all the fields then make a service to get all the data from the database and then make a component to display the data
export class GameReviewsView{
  id_review:number;
  rating:number;
  comment:string;
  username:string;
  id_game:number;
  game_title:string;
  game_genre:string;
  constructor(
    id_review:number,
    rating:number,
    comment:string,
    username:string,
    id_game:number,
    game_title:string,
    game_genre:string,
  ){
    this.id_review=id_review;
    this.rating=rating;
    this.comment=comment;
    this.username=username;
    this.id_game=id_game;
    this.game_title=game_title;
    this.game_genre=game_genre;
  }
}
